src/fu-engine.c

```diff
@@ -3045,32 +3117,33 @@ fu_engine_update_metadata_bytes (FuEngine *self, const gchar *remote_id,
 
 	/* verify file */
 	keyring_kind = fwupd_remote_get_keyring_kind (remote);
-	if (keyring_kind != FWUPD_KEYRING_KIND_NONE) {
-		g_autoptr(FuKeyring) kr = NULL;
-		g_autoptr(FuKeyringResult) kr_result = NULL;
-		g_autoptr(FuKeyringResult) kr_result_old = NULL;
+	if (keyring_kind == FWUPD_KEYRING_KIND_GPG ||
```

Comment:
I have to wonder - do we really still need to specify the type of keyring in the remote?  Why not just let the libjcat code look at it and determine from the stream what we have?

Reply:
That **is** a good question. Something like looking for `PKCS-7` or `GPG` in the first line of the signature?

Reply:
Yeah, I don't know for sure what kind of header there is for both if that's enough, but I would think it's entirely possible to tell from the stream what you have.

Reply:
I guess we certainly still need to say whether the remote needs to be signed or not, still, right?

Reply:
Yeah

Reply:
Also, depending on the answer to this next question, we could get rid of a lot of code: "_how much do we care about non-LVFS generated signed remotes_"? I believe `local.py` generates unsigned remotes, and I don't know of anyone that's deployed a private LVFS instance internally.

Reply:
Maybe it's worth putting a call out on social media to check? My gut is that you're right though.

Reply:
Quite a few people using complete LVFS mirrors, a a smattering of people using local.py it seems.